### PR TITLE
Add muc/omemo support

### PR DESCRIPTION
**Pull request by jelmer**
From @ichthyx in #40:

This PR add support for MUC (groupchat) and OMEMO (bot to user) and MUC OMEMO.

Currently private discussion (from user to bot) doesn't support omemo, we should implement omemo support in message() function. Not sure if I have the time for do it now, maybe later.

It add a new dependency which is slixmpp-omemo. Most of the code come from the slixmpp example : https://codeberg.org/poezio/slixmpp-omemo/src/branch/main/examples/echo_client.py

I used black to format code, it seems to have mess a lot of code, if you can suggest your way of formatting.

### CODE REVIEW

**Comment by jelmer** [comment]
Is there any chance you could submit the changes to introduce MUCs and the omemo support separately?

File: prometheus_xmpp/__main__.py [hunk 1]
```diff
@@ -179,14 +232,125 @@ def message(self, msg):
...
+            )
+            muc_affiliation = owners + admins + members + outcasts
+            recipients = [JID(nick) for nick in muc_affiliation]
+            recipients.remove(self.jid)
+            logging.debug("sending encrypted message to recipients %s", recipients)
+            await self.send_encrypted(mto=recipients, mtype="groupchat", body=mbody)
+
+    async def send_encrypted(self, mto: JID, mtype: Literal['chat', 'error', 'groupchat', 'headline', 'normal'], body):
+        """Helper to reply with encrypted messages"""
+        if self.muc:
```
Comment (jelmer) [inline prometheus_xmpp/__main__.py:260]: I have trouble reading this function. It seems like it just ignores mto if self.muc is set?

File: prometheus_xmpp/__main__.py [hunk 2]
```diff
@@ -179,14 +232,125 @@ def message(self, msg):
                 response = "Unknown command: %s" % args[0].lower()
             msg.reply(response).send()
 
+    async def dispatch_message(self, mto, mbody, mhtml):
```
Comment (jelmer) [inline prometheus_xmpp/__main__.py:235]: it seems this function ignores mto completely if self.muc is set, which is confusing; cam we just pass in self.muc_jid ?

**Comment by ichthyx** [comment]
>it seems this function ignores mto completely if self.muc is set, which is confusing; cam we just pass in self.muc_jid ?
Yeah I should review this, probably got a bit too fast using the slixmpp example.

>Is there any chance you could submit the changes to introduce MUCs and the omemo support separately?
What do you mean exactly ? One PR for MUCs support, and one PR for OMEMO ?

**Comment by jelmer** [comment]
> > Is there any chance you could submit the changes to introduce MUCs and the omemo support separately?
> 
> What do you mean exactly ? One PR for MUCs support, and one PR for OMEMO ?

Yeah - I'm not sure how easy or hard this would be so feel free to push back, but it would make reviewing the changes easier at least.

**Pull request closed without merging**